File: 2025_09/leetcode2025-09-15.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        int res = 0, n = text.size();
        unordered_map<char, int> broken;
        for (auto& c : brokenLetters) broken[c] = 1;
        int cnt = 0;
        for (int i=0; i<n; i++){
            cnt += broken[text[i]];
            if (text[i] == ' ' || i == n-1){
                if (cnt == 0){
                    res++;
                }
                cnt = 0;
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    string text = "hello world", brokenLetters = "ad";
    cout << s.canBeTypedWords(text, brokenLetters) << endl;
}

File: 2025_09/leetcode2025-09-15.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        for i in brokenLetters:
            text = text.replace(i, "*")
        return sum("*" not in w for w in text.split())

if __name__ == "__main__":
    s = Solution()
    text = "hello world"
    brokenLetters = "ad"
    print(s.canBeTypedWords(text, brokenLetters))
